node/docs/findingfiles.md

```diff
@@ -0,0 +1,98 @@
+   - Note, although multiple files can be resolved from a single match pattern, some scenarios may be too complex for a single pattern.
+
+2. Use a `filePath` input to specify a root directory or default-root directory, followed by a `multiLine` input for match patterns, when the following conditions are true:
+ - The task is designed to resolve multiple files/directories most of the time.
+ - OR the task is designed to resolve a single input most of the time, but adding a second instance of the task is unacceptable for scenarios that require resolving multiple files.
+   - Again, although multiple files can be resolved from a single match pattern, some scenarios may be too complex for a single pattern.
+
+## Task Lib Functions
+
+The above UI experiences translate to one of the following consumption patterns of the [task lib API](vsts-task-lib.md):
+
+1. `filePath` input only
+ - Call [findMatch](vsts-task-lib.md#taskfindMatch) and pass the filePath input as the pattern.
+
+2. `filePath` input to specify a root directory, followed by a `multiLine` input for match patterns.
+ - Call [find](vsts-task-lib.md#taskfind) to recursively find all paths under the specified root directory.
+ - Then call [match](vsts-task-lib.md#taskmatch) to filter the results using the multiLine input as the patterns.
+
+3. `filePath` input to specify a *default* root directory to root any unrooted patterns, followed by a `multiLine` input for match patterns.
+ - Call [findMatch](vsts-task-lib.md#taskfindMatch) and pass the filePath input as the defaultRoot and the multiLine input as the patterns.
+
+Note, use [getDelimitedInput](vsts-task-lib.md#taskgetDelimitedInput) to split a multiLine input using the delimiter `'\n'`.
+
+## Recommended FindOptions and MatchOptions
+
+The find and match functions apply the below defaults when the options parameters are not specified.
+
+### FindOptions
+The recommended defaults for FindOptions are below. Following soft links is generally appropriate unless deleting files.
+
+```javascript
+<FindOptions>{
+    followSpecifiedSymbolicLink = true,
+    followSymbolicLinks = true
+}
+```
+
+### MatchOptions
+The recommended defaults for MatchOptions are below. Supported pattern syntax is discussed further below in detail.
+
+```javascript
+<MatchOptions>{
+    debug = false,
+    nobrace = true,     // brace expansion off - brace expansion cannot be escaped on Windows
+    noglobstar = false, // globstar on
+    dot = true,         // make * match files that start with . without requiring an additional
+                        // pattern .* to match files that start with .
+    noext = false,      // extended globbing on
+    nocase = process.platform == 'win32' // case insensitive on Windows, otherwise case sensitive
+    nonull = false,
+    matchBase = false,
+    nocomment = false,  // support comments
+    nonegate = false,   // support negate pattern
+    flipNegate = false
+}
+```
+
+## Pattern Syntax
```

Comment:
I think a separate doc makes sense for user docs. For example, in the user docs we won't mention brace expansion at all because it will be off for in the box tasks.